src/util/json.cpp

```diff
@@ -162,3 +164,48 @@ void jsont::swap(jsont &other)
   other.object.swap(object);
   other.value.swap(value);
 }
+
+bool operator==(const jsont &left, const jsont &right)
+{
+  if(left.kind != right.kind)
+    return false;
+  switch(left.kind)
+  {
+  case jsont::kindt::J_NULL:
+    return true;
+  case jsont::kindt::J_TRUE:
+    return true;
+  case jsont::kindt::J_FALSE:
+    return true;
+  case jsont::kindt::J_NUMBER:
+    return left.value == right.value;
+  case jsont::kindt::J_STRING:
+    return left.value == right.value;
```

Comment:
Overall design question: shouldn't we do this more in a visitor-style where each call deriving from `jsont` defines each own `operator==` and the switch-case here would only do the downcasts and then invoke the more specific equality operators?

Reply:
Yes, except `jsont` much like `irept` doesn't allow any virtual functions. Therefore the visitor pattern is not applicable.

Reply:
Of course, hence me saying "visitor-style." But also my comment was intended as a thought, not any kind of hard requirement.